### ISSUE TITLE
Tagging is not applied by S3.upload() for large files

### ISSUE THREAD

**Issue by damonmaria** [aws/aws-sdk-js#1412]
If I upload a small file using `S3.upload()` and specify tagging then the tags are applied. If however, the file is larger than 15M and under the hood `upload` uses a multipart upload then the tags are silently not applied. 

### Example
```js
const AWS = require('aws-sdk')
const fs = require('fs')

const s3 = new AWS.S3({
  region: 'ap-southeast-2',
  credentials: new AWS.SharedIniFileCredentials(),
})

s3.upload(
  {
    Bucket: 'vigil.media.dev-damon',
    Key: 'test.upload.big',
    Tagging: 'someKey=someValue',
    Body: fs.createReadStream('./some-big.file'),
  },
  (err, data) => {
    if (err) {
      console.errro(err)
    } else {
      console.dir(data)
    }
  }
)
```

And then check if tags applied:
```bash
$ aws s3api get-object-tagging --bucket vigil.media.dev-damon --region ap-southeast-2 --key test.upload.big
{
    "TagSet": []
}
```

If `./some-big.file` is changed to something less than 15M then checking the tags afterwards show they have been applied. 

### Environment
AWS SDK 2.28.0
Node v7.7.3
MacOS 10.12.3

**Comment by chrisradek** [comment]
@damonmaria 
It looks like S3 supports tagging a multi-part upload only after it has finished uploading. I think it should be safe for us to call [`putObjectTagging`](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#putObjectTagging-property) after a multi-part upload, though that will mean an extra request.

**Comment by damonmaria** [comment]
@chrisradek Although an extra request it is a multipart upload so is already multiple requests. When uploading a large file you're not expecting millisecond latency.

**Comment by chrisradek** [comment]
@damonmaria 
Agreed. I've marked this as a bug. Our plan is to make the extra request if upload does a multi-part upload. I'll comment here once we have a PR up.

**Comment by gabegorelick** [comment]
Is this documented anywhere? Judging by [the linked PR](https://github.com/aws/aws-sdk-js/pull/1425), callers should be using this like so
```js
s3.upload(
    {
        Bucket: 'BUCKET',
        Key: 'KEY',
        Body: new Buffer('some data')
    },
    {
        tags: [
            {
                Key: 'key',
                Value: 'value',
            },
            {
                Key: 'otherKey',
                Value: 'otherValue',
            },
        ],
    }
);
```

instead of passing a `Tagging` string option in the first argument to `upload`. However, this is not documented in the [JS SDK docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#upload-property).

**Comment by liquidaty** [comment]
It appears that it should be possible to put it directly into params (though I haven't scrutinized AWS.util.merge() enough to say for sure). The signature accepts either (params, callback) or (params, options, callback); however, the latter seems to merge into the former before the eventual unified call to ManagedUpload:

```
  // aws-sdk.js upload function:
  upload: function upload(params, options, callback) {
    if (typeof options === 'function' && callback === undefined) {
      callback = options;
      options = null;
    }

    options = options || {};
    options = AWS.util.merge(options || {}, {service: this, params: params});

    var uploader = new AWS.S3.ManagedUpload(options);
    if (typeof callback === 'function') uploader.send(callback);
    return uploader;
  }
});
```
